src/adapters/scene.js

```diff
@@ -107,28 +107,59 @@ Scene.prototype.initMapBox = function() {
         this.mb.getCanvas().style.cursor = '';
       });
 
+      // The OSM PoIs are on separate interactive layers.
+      // This code listens to clicks on each interactive layer, and if an interactive PoI marker is clicked, instantiate a MapPoi and save it in e._mapPoi.
+      // The global click listener (below) will open the corresponding PoI panel or complete the direction panel fields with it.
       this.mb.on('click', interactiveLayer, async e => {
         e._interactiveClick = true;
         if (e.features && e.features.length > 0) {
-          const mapPoi = new MapPoi(e.features[0]);
-          window.app.navigateTo(`/place/${toUrl(mapPoi)}`, { poi: mapPoi });
+          e._mapPoi = new MapPoi(e.features[0]);
         }
       });
 
       this.popup.addListener(interactiveLayer);
     });
 
+    // This code listens to clicks on the whole map.
+    // - On mobile: return to home if no OSM PoI was clicked and Direction panel is not open
+    // - If an OSM PoI was clicked, use it.
+    // - Otherwise, create a latlon PoI.
+    // Then:
+    // - If the Direction panel is open, complete itinerary fields with the current PoI (QMAPS-1205)
+    // - Otherwise: show a PoI panel
     this.mb.on('click', e => {
-      if (!e._interactiveClick && Device.isMobile()
-          && !document.querySelector('.directions-open')) {
+
+      // Mobile
+      if (Device.isMobile() && !e._mapPoi && !document.querySelector('.directions-open')) {
         window.app.navigateTo('/');
-      }
-      // Disable POI anywhere feature on mobile until we opt for an adapted UX
-      if (Device.isMobile() || e._interactiveClick || this.routeDisplayed) {
         return;
       }
-      const poi = new LatLonPoi(e.lngLat);
-      window.app.navigateTo(`/place/${toUrl(poi)}`, { poi });
+
+      // Select PoI (OSM or LatLon)
+      let poi;
+      if (e._mapPoi) {
+        poi = e._mapPoi;
+      } else {
+        poi = new LatLonPoi(e.lngLat);
+      }
+
+      // Complete Direction panel if it's open, open PoI panel if it's already full
+      if (document.querySelector('.directions-open')) {
```

Comment:
The logic looks good, but it would be better to move it where the data state of the form is maintained, i.e. the `DirectionPanel` itself.
Here you need to refer directly to DOM elements, which requires to put fine knowledge of how the UI layer works in that file that isn't related to routes.

A better solution would be to send a single type of event, with no explicit `origin` or `destination` indication, and let `DirectionPanel` decide which field to fill based only on its internal state.

Among added benefits, it would make it more robust if we add more than two points in the future.